deepcell_spots/applications/polaris.py

```diff
@@ -23,227 +23,132 @@
-            preprocess_kwargs (dict): Keyword arguments to pass to the
-                pre-processing function.
-            postprocess_kwargs (dict): Keyword arguments to pass to the
-                post-processing function.
-        Raises:
-            ValueError: Input data must match required rank, calculated as one
-                dimension more (batch dimension) than expected by the model.
-            ValueError: Input data must match required number of channels.
-        Returns:
-            numpy.array: Coordinate spot locations
-        """
-        # Check input size of image
-        if len(image.shape) != self.required_rank:
-            raise ValueError('Input data must have {} dimensions. '
-                             'Input data only has {} dimensions'.format(
-                                 self.required_rank, len(image.shape)))
-
-        if image.shape[-1] != self.required_channels:
-            raise ValueError('Input data must have {} channels. '
-                             'Input data only has {} channels'.format(
-                                 self.required_channels, image.shape[-1]))
-
-        # Resize image, returns unmodified if appropriate
-        resized_image = self._resize_input(image, image_mpp)
-
-        # Generate model outputs
-        output_images = self._run_model(
-            image=resized_image, batch_size=batch_size,
-            pad_mode=pad_mode, preprocess_kwargs=preprocess_kwargs
-        )
-
-        # Resize output_images back to original resolution if necessary
-        label_image = self._resize_output(output_images, image.shape)
+    def __init__(self,
+                 segmentation_model=None,
+                 spots_model=None):
+        if segmentation_model is None:
+            model_path = '../notebooks/models/CytoplasmSegmentation'
+            segmentation_model = tf.keras.models.load_model(model_path)
 
-        # Postprocess predictions to create label image
-        predicted_spots = self._postprocess(label_image, **postprocess_kwargs)
+        self.spots_app = SpotDetection(model=spots_model)
+        self.segmentation_app = CytoplasmSegmentation(model=segmentation_model)
 
-        return predicted_spots
+        self.segmentation_app.preprocessing_fn = histogram_normalization
+        self.segmentation_app.postprocessing_fn = deep_watershed
 
     def predict(self,
                 image,
-                batch_size=4,
                 image_mpp=None,
-                pad_mode='reflect',
-                preprocess_kwargs=None,
-                postprocess_kwargs=None,
-                threshold=0.9):
-        """Generates a list of coordinate spot locations of the input
-        running prediction with appropriate pre and post processing
-        functions.
+                # segmentation
+                cytoplasm_channel=0,
+                # spots
+                spots_channel=1,
+                threshold=0.95,
```

Comment:
Would `spots_threshold` make sense? It would make it more clear which model the parameter belongs to? Same thing with `clip`